pkg/kubernetes/topology.js

```diff
@@ -72,16 +76,26 @@ define([
                             var addresses = subset.addresses || [ ];
                             addresses.forEach(function(address) {
                                 if (address.targetRef && address.targetRef.kind == "Pod")
-                                    pods["Pod:" + address.targetRef.uid] = { };
+                                    rels["Pod:" + address.targetRef.uid] = { };
                             });
                         });
 
                     /* For ReplicationControllers we just do the selection ourselves */
                     } else if (item.kind === "ReplicationController") {
-                        pods = client.select("Pod", item.metadata.namespace, item.spec.selector || { });
+                        rels = client.select("Pod", item.metadata.namespace, item.spec.selector || { });
+
+                    } else if (item.kind === "DeploymentConfig") {
+                        rels = client.select("ReplicationController", item.metadata.namespace,
+                                             {"openshift.io/deployment-config.name" : item.metadata.name});
+                    /* For Routes just build it out */
+                    } else if (item.kind === "Route" && item.spec.to) {
+                        var rel = client.lookup(item.spec.to.kind,
+                                                item.spec.to.name,
+                                                item.metadata.namespace);
+                        if (rel)
+                            rels[item.spec.to.kind + ":" + rel.metadata.uid] = rel;
```

Comment:
We now set a .key attribute on the items retrieved from the API server. Does that help here?